doom-modeline-segments.el

```diff
@@ -846,19 +868,33 @@ mouse-2: Show help for minor mode")
               (pcase status
                 ('finished  (when flycheck-current-errors
                               (let-alist (doom-modeline--flycheck-count-errors)
-                                (if doom-modeline-check-simple-format
-                                    (doom-modeline-check-text
-                                     (number-to-string (+ .error .warning .info))
-                                     (cond ((> .error 0) 'doom-modeline-urgent)
-                                           ((> .warning 0) 'doom-modeline-warning)
-                                           (t 'doom-modeline-info)))
-                                  (format "%s/%s/%s"
-                                          (doom-modeline-check-text (number-to-string .error)
-                                                                      'doom-modeline-urgent)
-                                          (doom-modeline-check-text (number-to-string .warning)
-                                                                      'doom-modeline-warning)
-                                          (doom-modeline-check-text (number-to-string .info)
-                                                                      'doom-modeline-info))))))
+                                (pcase doom-modeline-check-simple-format
+                                  ('t (doom-modeline-check-text (number-to-string (+ .error .warning .info))
+                                                                (cond ((> .error 0) 'doom-modeline-urgent)
+                                                                      ((> .warning 0) 'doom-modeline-warning)
+                                                                      (t 'doom-modeline-info))))
+                                  ('nil (format "%s/%s/%s"
+                                                (doom-modeline-check-text (number-to-string .error)
+                                                                          'doom-modeline-urgent)
+                                                (doom-modeline-check-text (number-to-string .warning)
+                                                                          'doom-modeline-warning)
+                                                (doom-modeline-check-text (number-to-string .info)
+                                                                          'doom-modeline-info)))
+                                  ('icons (format "%s %s %s %s %s"
+                                                  (doom-modeline-check-text (number-to-string .error)
```

Comment:
Should handle the error icon here.

Reply:
I don't think so, found this way better, the icon error is already provided by [flymake|flycheck]-icon, which provides using their menu, however if it's required adding the error icon i would gladly do it.

Reply:
The behaviors are different. the current error icon in `doom-modeline-update-[flymake|flycheck]-icon` may be different  color. That's why I feel the implement is not perfect.

Reply:
Oh, you don't have to worrry about that, this line [933](https://github.com/seagle0128/doom-modeline/pull/711/files#diff-0505629a410f96e739a924cc71c2ef537a25d27a64d97615e8e7ae3d06906130R993) automatically don't change icon color if `doom-modeline-check-simple-format` is set to `icons` only change to 1 color `doom-modeline-urgent`.

This doesn't interfere with other check formats.

| `'nil`  | `'icons`  | `'t`  |
|---|---|---|
| ![3Sin título](https://github.com/seagle0128/doom-modeline/assets/163072436/c151f451-749e-489b-8b45-6bdfa03391c3)  | ![1Sin título](https://github.com/seagle0128/doom-modeline/assets/163072436/2d5d679d-74b5-4e9a-a4e7-56b2d76190e9)  |  ![2Sin título](https://github.com/seagle0128/doom-modeline/assets/163072436/9ec9416b-77ad-4014-aa89-400cd6bff564) |

Reply:
Did you observe the extra spaces after the icons? In the screenshots, no space after the error icon.

Reply:
Thank you, that was the solution i was trying to find for those extra spaces.

Now this should be fixed.
![1Sin título](https://github.com/seagle0128/doom-modeline/assets/163072436/a7dce427-e56a-40ef-b8d3-99094db39733)

Reply:
Thanks. It works now. But, the help-echo and key-map are not correct yet. I am thinking it's a design issue.